shared/js/advanced_telemetry_helper.js

```diff
@@ -0,0 +1,154 @@
+/*
+ * This module is a helper for recording telemetry data in Gecko’s telemetry
+ * system. Currently, only data from certified apps will be recorded.
+ *
+ * Usage Example:
+ *  var ath = new AdvancedTelemetryHelper();
+ *  // DO SOMETHING
+ *  // This creates and increments the counter on this metric called
+ *  // 'mycountmetric'
+ *  ath.count('mycountmetric'); // Count will be 1
+ *  ath.count('mycountmetric'); // Count is now 2
+ *  // DO SOMETHING
+ *  // This creates and logs a linear histogram called 'mylinearmetric' with
+ *  // value 100, minimum of 0, max of 1000, with 10 buckets.
+ *  ath.linear('mylinearmetric', 100, 0, 1000, 10);
```

Comment:
I thought about this and my thoughts were that this will be similar to a logging api for a developer.  Essentially, you don't want to use multiple lines to log.  Then the initialization becomes an extra line of code for the developer.  So, I opted to keep it thin.  Let me know if you want to talk more about this as there may be some points I'm missing.